### ISSUE TITLE
Configs Override For CLI-Injected Workloads

### ISSUE THREAD

**Issue by ihcsim** [linkerd/linkerd2#2590]
This is a `stable-2.4` follow-up task for https://github.com/linkerd/linkerd2/issues/2498#issuecomment-477773754. In `stable-2.3`, we have decided to postpone the support of overriding configs for existing CLI-injected workloads, due to reasons described in the above comment and #2576. 

The focus of this task is to ensure that following #2576, the configs override of existing CLI-injected workloads work. Depending on the implementation decision of #2576, if the proxy spec is removed from the output of `linkerd inject`, then no changes to the webhook will be necessary, as the webhook will pick up the pod (during `CREATE`) and mutate its pod spec to include the proxy container.

**Comment by ihcsim** [comment]
Depends on #2710 #2711 #2715

**Comment by ihcsim** [comment]
Fixed by #2710 #2711 #2715
